![example](images/director_shot.jpeg)

# Project Title

**Authors:** Victor Kang
***

## Overview

A one-paragraph overview of the project, including the business problem, data, methods, results and recommendations.

## Business Problem

Microsoft has been suffering from a severe case of FOMO (Fear-Of-Missing-Out) as they've watched many of their competitors succeed in opening up their own movie studios and creating original video content! Microsoft wants in on the action and is now opening their own movie studio! I've been tasked with providing critical data-based market research to ensure that Microsoft's first movie will be a global success!

***
#### Primary Objectives and Qualifications:
1. Explore and analyze what **types** of films are **currently** doing the **best at the box office**.
* The key terms here in bold must first be defined. For this Project, they shall be defined as follows:
>* **Types**: There are many ways to classify or categorize films, most common being by Genre. We can also categorize movies by their budget range, ie. big budget vs small budget.
>* **Currently**: Because Microsoft asked for "currently", we know we should only consider modern movies in our upcoming analysis. Exactly how modern will be influenced by our available data. Specific Date Range To Be Determined! But we should at minimum aim to include movies released this year (2022)!
>* **Best at the Box Office**: "Best" will be defined solely by the financial performance of movies at the Worldwide Box Office. To measure financial performance, we will explore the Worldwide Box Office Gross of movies and compare it to their Production Budgets to calculate the *Profit/Loss* and *Return-On-Investment* metrics.  

* In short, our first objective is to determine which Genres and Budget Ranges of modern movies have produced the highest profit and return-on-investment for their movie studios! 

2. Provide (3) actionable insights / concrete business recommendations based on the analysis. 
* We plan to provide budget range recommendations and how budgets could have a relationship to financial success.
* Genre
* Recommendations for cast and crew! Actor, Actress, Director, and Writer recommendations.

* What are the business's pain points related to this project?
* How did you pick the data analysis question(s) that you did?
* Why are these questions important from a business perspective?
***

## Data Understanding

Describe the data being used for this project.
***
Questions to consider:
* Where did the data come from, and how do they relate to the data analysis questions?
* What do the data represent? Who is in the sample and what variables are included?
* What is the target variable?
* What are the properties of the variables you intend to use?

***

## Prelimary Data Exploration

Provided Data for Project:
* **Box Office Mojo**, compressed CSV file
* **Rotten Tomatoes**, 2 compressed TSV files
* **TheMovieDB**, compressed CSV file
* **The Numbers**, compressed CSV file
* **IMDB**, SQLlite database

We will begin by exploring each of the provided data sets to examine if they will relate to our data analyis objectives. The target variables we hope to find are data on movie titles, genres, years, budgets, box office performance, cast, and crew!
We would expect that genre, cast, and crew to be string variables. We hope we will find year, budgets, and box office performance data to be some type of numerical variable, such as a float.

After exploring each provided data set, if we don't find what we need, we may need to explore other sources for supplemental data! Let's see! 
First, let's prepare our notebook by importing our standard packages.

In [1]:
# Import standard packages
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import sqlite3

%matplotlib inline

The first data set we will explore is the **Box Office Mojo** data set.

In [8]:
#Use pandas to load the data
BoxOfficeMojo = pd.read_csv("zippedData/bom.movie_gross.csv.gz")
BoxOfficeMojo

title      studio  domestic_gross  \
0                                     Toy Story 3          BV     415000000.0   
1                      Alice in Wonderland (2010)          BV     334200000.0   
2     Harry Potter and the Deathly Hallows Part 1          WB     296000000.0   
3                                       Inception          WB     292600000.0   
4                             Shrek Forever After        P/DW     238700000.0   
...                                           ...         ...             ...   
3382                                    The Quake       Magn.          6200.0   
3383                  Edward II (2018 re-release)          FM          4800.0   
3384                                     El Pacto        Sony          2500.0   
3385                                     The Swan  Synergetic          2400.0   
3386                            An Actor Prepares       Grav.          1700.0   

     foreign_gross  year  
0        652000000  2010  
1        691300000  2010  
2        664300000  2010  
3        535700000  2010  
4        513900000  2010  
...            ...   ...  
3382           NaN  2018  
3383           NaN  2018  
3384           NaN  2018  
3385           NaN  2018  
3386           NaN  2018  

[3387 rows x 5 columns]

At first glance, we see that this data set has potentially useful box office performance data for 3,387 movies. However, we also suspect that this data set only contains movies from 2010-2018 due to what we see in the year column! Let's confirm if that's the case.

In [9]:
#Use the .describe() method to get a quick summary of the data
BoxOfficeMojo.describe()

domestic_gross         year
count    3.359000e+03  3387.000000
mean     2.874585e+07  2013.958075
std      6.698250e+07     2.478141
min      1.000000e+02  2010.000000
25%      1.200000e+05  2012.000000
50%      1.400000e+06  2014.000000
75%      2.790000e+07  2016.000000
max      9.367000e+08  2018.000000

The minimum and maximum values for the year column confirm our suspicion. Not good. 2018 is just not recent enough for our objective. We may need to find outside sources to update this data set with more recent data, but for now, let's take note that we could use this Box Office Mojo data set.
***

Next, let's explore the (2) **Rotten Tomatoes** data sets.

In [17]:
#Use pandas to load the data
pd.read_csv("zippedData/rt.movie_info.tsv.gz", delimiter='\t')

id                                           synopsis rating  \
0        1  This gritty, fast-paced, and innovative police...      R   
1        3  New York City, not-too-distant-future: Eric Pa...      R   
2        5  Illeana Douglas delivers a superb performance ...      R   
3        6  Michael Douglas runs afoul of a treacherous su...      R   
4        7                                                NaN     NR   
...    ...                                                ...    ...   
1555  1996  Forget terrorists or hijackers -- there's a ha...      R   
1556  1997  The popular Saturday Night Live sketch was exp...     PG   
1557  1998  Based on a novel by Richard Powell, when the l...      G   
1558  1999  The Sandlot is a coming-of-age story about a g...     PG   
1559  2000  Suspended from the force, Paris cop Hubert is ...      R   

                                                  genre            director  \
0                   Action and Adventure|Classics|Drama    William Friedkin   
1                     Drama|Science Fiction and Fantasy    David Cronenberg   
2                     Drama|Musical and Performing Arts      Allison Anders   
3                            Drama|Mystery and Suspense      Barry Levinson   
4                                         Drama|Romance      Rodney Bennett   
...                                                 ...                 ...   
1555   Action and Adventure|Horror|Mystery and Suspense                 NaN   
1556                 Comedy|Science Fiction and Fantasy        Steve Barron   
1557  Classics|Comedy|Drama|Musical and Performing Arts      Gordon Douglas   
1558    Comedy|Drama|Kids and Family|Sports and Fitness  David Mickey Evans   
1559  Action and Adventure|Art House and Internation...                 NaN   

                                                writer  theater_date  \
0                                       Ernest Tidyman   Oct 9, 1971   
1                         David Cronenberg|Don DeLillo  Aug 17, 2012   
2                                       Allison Anders  Sep 13, 1996   
3                      Paul Attanasio|Michael Crichton   Dec 9, 1994   
4                                         Giles Cooper           NaN   
...                                                ...           ...   
1555                                               NaN  Aug 18, 2006   
1556  Terry Turner|Tom Davis|Dan Aykroyd|Bonnie Turner  Jul 23, 1993   
1557                                               NaN   Jan 1, 1962   
1558                  David Mickey Evans|Robert Gunter   Apr 1, 1993   
1559                                        Luc Besson  Sep 27, 2001   

          dvd_date currency  box_office      runtime             studio  
0     Sep 25, 2001      NaN         NaN  104 minutes                NaN  
1      Jan 1, 2013        $     600,000  108 minutes  Entertainment One  
2     Apr 18, 2000      NaN         NaN  116 minutes                NaN  
3     Aug 27, 1997      NaN         NaN  128 minutes                NaN  
4              NaN      NaN         NaN  200 minutes                NaN  
...            ...      ...         ...          ...                ...  
1555   Jan 2, 2007        $  33,886,034  106 minutes    New Line Cinema  
1556  Apr 17, 2001      NaN         NaN   88 minutes  Paramount Vantage  
1557  May 11, 2004      NaN         NaN  111 minutes                NaN  
1558  Jan 29, 2002      NaN         NaN  101 minutes                NaN  
1559  Feb 11, 2003      NaN         NaN   94 minutes  Columbia Pictures  

[1560 rows x 12 columns]

At first glance, we see some columns that we could use for classification, such as genre, rating, director, writer, etc. There's even box office data that is one of our target variables. Unfortunately, we conclude that this data set is not useful because the movie titles are not identified. Each record has an id column with numerical values, but we don't have a key to know what that means. Thus, we will not use this data.

In [19]:
#Use pandas to load the data. Note: define the encoding or else you get an error.
pd.read_csv("zippedData/rt.reviews.tsv.gz", delimiter='\t', encoding='latin1')

id                                             review rating   fresh  \
0         3  A distinctly gallows take on contemporary fina...    3/5   fresh   
1         3  It's an allegory in search of a meaning that n...    NaN  rotten   
2         3  ... life lived in a bubble in financial dealin...    NaN   fresh   
3         3  Continuing along a line introduced in last yea...    NaN   fresh   
4         3             ... a perverse twist on neorealism...     NaN   fresh   
...     ...                                                ...    ...     ...   
54427  2000  The real charm of this trifle is the deadpan c...    NaN   fresh   
54428  2000                                                NaN    1/5  rotten   
54429  2000                                                NaN    2/5  rotten   
54430  2000                                                NaN  2.5/5  rotten   
54431  2000                                                NaN    3/5   fresh   

                  critic  top_critic         publisher                date  
0             PJ Nabarro           0   Patrick Nabarro   November 10, 2018  
1         Annalee Newitz           0           io9.com        May 23, 2018  
2           Sean Axmaker           0  Stream on Demand     January 4, 2018  
3          Daniel Kasman           0              MUBI   November 16, 2017  
4                    NaN           0      Cinema Scope    October 12, 2017  
...                  ...         ...               ...                 ...  
54427      Laura Sinagra           1     Village Voice  September 24, 2002  
54428  Michael Szymanski           0        Zap2it.com  September 21, 2005  
54429       Emanuel Levy           0   EmanuelLevy.Com       July 17, 2005  
54430   Christopher Null           0    Filmcritic.com   September 7, 2003  
54431    Nicolas Lacroix           0      Showbizz.net   November 12, 2002  

[54432 rows x 8 columns]

This second Rotten Tomatoes data set has the same problem as the previous set. We don't have a way to identify the movie title for each record. Also, this dataset's focus is on review and critic information. We are not interested in that kind of  data for our objective. Our Studio has made it clear that they only care about financial performance and could care less about if a movie is well-received by critics.
***

Let's explore the next data set. **The Numbers**.

In [27]:
#Use pandas to load the data
TheNumbers = pd.read_csv("zippedData/tn.movie_budgets.csv.gz")
TheNumbers

id  release_date                                        movie  \
0      1  Dec 18, 2009                                       Avatar   
1      2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2      3   Jun 7, 2019                                 Dark Phoenix   
3      4   May 1, 2015                      Avengers: Age of Ultron   
4      5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   
...   ..           ...                                          ...   
5777  78  Dec 31, 2018                                       Red 11   
5778  79   Apr 2, 1999                                    Following   
5779  80  Jul 13, 2005                Return to the Land of Wonders   
5780  81  Sep 29, 2015                         A Plague So Pleasant   
5781  82   Aug 5, 2005                            My Date With Drew   

     production_budget domestic_gross worldwide_gross  
0         $425,000,000   $760,507,625  $2,776,345,279  
1         $410,600,000   $241,063,875  $1,045,663,875  
2         $350,000,000    $42,762,350    $149,762,350  
3         $330,600,000   $459,005,868  $1,403,013,963  
4         $317,000,000   $620,181,382  $1,316,721,747  
...                ...            ...             ...  
5777            $7,000             $0              $0  
5778            $6,000        $48,482        $240,495  
5779            $5,000         $1,338          $1,338  
5780            $1,400             $0              $0  
5781            $1,100       $181,041        $181,041  

[5782 rows x 6 columns]

This data set looks promising! We see movie titles and box office performance data for 5,782 movies! This is the first time we've seen a key variable present that will be critical to our objectives; production budget data! Let's take a quick look at the date range of the data.

In [134]:
#Use the last 4 characters from release_date column to create a new year column as an integer data type!
TheNumbers['year'] = TheNumbers['release_date'].astype(str).str[-4:].astype(int)

#Use .describe() method to see year range
TheNumbers.describe()

id         year
count  5782.000000  5782.000000
mean     50.372363  2003.967139
std      28.821076    12.724386
min       1.000000  1915.000000
25%      25.000000  2000.000000
50%      50.000000  2007.000000
75%      75.000000  2012.000000
max     100.000000  2020.000000

In [135]:
#check what 2020 movies this data has
TheNumbers[TheNumbers['year'] == 2020]

id  release_date                   movie production_budget  \
194   95  Dec 31, 2020                Moonfall      $150,000,000   
535   36  Feb 21, 2020        Call of the Wild       $82,000,000   
1205   6  Dec 31, 2020  Hannibal the Conqueror       $50,000,000   

     domestic_gross worldwide_gross  year  
194              $0              $0  2020  
535              $0              $0  2020  
1205             $0              $0  2020

The dataset has data for movies from 1915 to 2020; however, upon further exploration of the 2020 movies, we see that the 2020 data is actually unreliable! If we first update this dataset to include current movie data, TheNumbers could be very useful!

***
Let's explore the next data set. **TheMovieDB**.

In [23]:
#Use pandas to load the data. Set index column to 0!
pd.read_csv("zippedData/tmdb.movies.csv.gz", index_col=0)

genre_ids      id original_language  \
0          [12, 14, 10751]   12444                en   
1      [14, 12, 16, 10751]   10191                en   
2            [12, 28, 878]   10138                en   
3          [16, 35, 10751]     862                en   
4            [28, 878, 12]   27205                en   
...                    ...     ...               ...   
26512             [27, 18]  488143                en   
26513             [18, 53]  485975                en   
26514         [14, 28, 12]  381231                en   
26515      [10751, 12, 28]  366854                en   
26516             [53, 27]  309885                en   

                                     original_title  popularity release_date  \
0      Harry Potter and the Deathly Hallows: Part 1      33.533   2010-11-19   
1                          How to Train Your Dragon      28.734   2010-03-26   
2                                        Iron Man 2      28.515   2010-05-07   
3                                         Toy Story      28.005   1995-11-22   
4                                         Inception      27.920   2010-07-16   
...                                             ...         ...          ...   
26512                         Laboratory Conditions       0.600   2018-10-13   
26513                               _EXHIBIT_84xxx_       0.600   2018-05-01   
26514                                  The Last One       0.600   2018-10-01   
26515                                  Trailer Made       0.600   2018-06-22   
26516                                    The Church       0.600   2018-10-05   

                                              title  vote_average  vote_count  
0      Harry Potter and the Deathly Hallows: Part 1           7.7       10788  
1                          How to Train Your Dragon           7.7        7610  
2                                        Iron Man 2           6.8       12368  
3                                         Toy Story           7.9       10174  
4                                         Inception           8.3       22186  
...                                             ...           ...         ...  
26512                         Laboratory Conditions           0.0           1  
26513                               _EXHIBIT_84xxx_           0.0           1  
26514                                  The Last One           0.0           1  
26515                                  Trailer Made           0.0           1  
26516                                    The Church           0.0           1  

[26517 rows x 9 columns]

While we see movie titles and years for each record, we don't see any  other useful data for our objectives. We don't have a key to translate the genre id's, and we don't care about review information. We won't be using TheMovieDB data.
***

Finally, let's explore the **IMDB** database! From our literature, we know that the ERD for this database is as follow:
![example](images/movie_data_erd.jpeg)

Immediately, we identify that we may be interested in the following:
* movie_basics table for movie_id, start_year, and genres
* persons table for person_id, and primary_name
* principals table for movie_id, ordering, person_id, category, job
* directors table for person_id, movie_id
* writers table for person_id, movie_id

**NOTE FOR READER: I spent much more time and lines of code exploring the SQL database using pandas, queries, joining various tables, extracting, filtering, aggregating, etc. but I have not included them here for this clean notebook because I ultimately decided against using the data for reasons below. If you desire, you may open the dirty notebook to see more exploratory code using SQL.**

In [136]:
# Establish connection to SQLlite database
# Unzip the 'im.db.zip' file first if not done so already

conn = sqlite3.connect("zippedData/im.db")

In [32]:
# Use pandas to confirm what tables are in database
pd.read_sql("""SELECT name FROM sqlite_master WHERE type = 'table';""", conn)

name
0   movie_basics
1      directors
2      known_for
3     movie_akas
4  movie_ratings
5        persons
6     principals
7        writers

In [111]:
#Let's first check out the movie_basics table to see what movies they got
movie_basics = pd.read_sql("""
SELECT *
FROM movie_basics
;
""", conn)

movie_basics

movie_id                                primary_title  \
0       tt0063540                                    Sunghursh   
1       tt0066787              One Day Before the Rainy Season   
2       tt0069049                   The Other Side of the Wind   
3       tt0069204                              Sabse Bada Sukh   
4       tt0100275                     The Wandering Soap Opera   
...           ...                                          ...   
146139  tt9916538                          Kuambil Lagi Hatiku   
146140  tt9916622  Rodolpho Teóphilo - O Legado de um Pioneiro   
146141  tt9916706                              Dankyavar Danka   
146142  tt9916730                                       6 Gunn   
146143  tt9916754               Chico Albuquerque - Revelações   

                                     original_title  start_year  \
0                                         Sunghursh        2013   
1                                   Ashad Ka Ek Din        2019   
2                        The Other Side of the Wind        2018   
3                                   Sabse Bada Sukh        2018   
4                             La Telenovela Errante        2017   
...                                             ...         ...   
146139                          Kuambil Lagi Hatiku        2019   
146140  Rodolpho Teóphilo - O Legado de um Pioneiro        2015   
146141                              Dankyavar Danka        2013   
146142                                       6 Gunn        2017   
146143               Chico Albuquerque - Revelações        2013   

        runtime_minutes                genres  
0                 175.0    Action,Crime,Drama  
1                 114.0       Biography,Drama  
2                 122.0                 Drama  
3                   NaN          Comedy,Drama  
4                  80.0  Comedy,Drama,Fantasy  
...                 ...                   ...  
146139            123.0                 Drama  
146140              NaN           Documentary  
146141              NaN                Comedy  
146142            116.0                  None  
146143              NaN           Documentary  

[146144 rows x 6 columns]

Over 146 thousand movies! That's alot! Let's see the date range of the movies!

In [36]:
#Use the .describe() method to get a quick summary of the data
movie_basics.describe()

start_year  runtime_minutes
count  146144.000000    114405.000000
mean     2014.621798        86.187247
std         2.733583       166.360590
min      2010.000000         1.000000
25%      2012.000000        70.000000
50%      2015.000000        87.000000
75%      2017.000000        99.000000
max      2115.000000     51420.000000

That's odd. The min year is 2010 but the max year is 2115?! Let's investigate further.

In [38]:
#use .value_counts() method on the year column to see how many movies per year
movie_basics['start_year'].value_counts()

2017    17504
2016    17272
2018    16849
2015    16243
2014    15589
2013    14709
2012    13787
2011    12900
2010    11849
2019     8379
2020      937
2021       83
2022       32
2023        5
2024        2
2027        1
2026        1
2025        1
2115        1
Name: start_year, dtype: int64

In [68]:
#sort the dataframe by Year, show the top 10 records
movie_basics.sort_values(by = 'start_year', ascending = False).head(10)

movie_id                              primary_title  \
89506    tt5174640                                  100 Years   
96592    tt5637536                                   Avatar 5   
2949    tt10300398                    Untitled Star Wars Film   
52213    tt3095356                                   Avatar 4   
105187   tt6149054  Fantastic Beasts and Where to Find Them 5   
2948    tt10300396                    Untitled Star Wars Film   
2483    tt10255736                    Untitled Marvel Project   
16337    tt1757678                                   Avatar 3   
2906    tt10298848        Untitled Disney Live-Action Project   
106865   tt6258542         Wraith of the Umbra and Eidolon II   

                                   original_title  start_year  \
89506                                   100 Years        2115   
96592                                    Avatar 5        2027   
2949                      Untitled Star Wars Film        2026   
52213                                    Avatar 4        2025   
105187  Fantastic Beasts and Where to Find Them 5        2024   
2948                      Untitled Star Wars Film        2024   
2483                      Untitled Marvel Project        2023   
16337                                    Avatar 3        2023   
2906          Untitled Disney Live-Action Project        2023   
106865         Wraith of the Umbra and Eidolon II        2023   

        runtime_minutes                    genres  
89506               NaN                     Drama  
96592               NaN  Action,Adventure,Fantasy  
2949                NaN                   Fantasy  
52213               NaN  Action,Adventure,Fantasy  
105187              NaN  Adventure,Family,Fantasy  
2948                NaN                      None  
2483                NaN                    Action  
16337               NaN    Action,Adventure,Drama  
2906                NaN                      None  
106865              NaN   Adventure,Drama,Fantasy

In [110]:
#use runtime data as an estimate to see how many of the movies have reliable data for years 2019-2022
#use a for loop to iterate years
#use .isnull() and .value_counts() to see how many of the records have no data

for year in [2018, 2019, 2020, 2021, 2022]:
    print('# of ', year, 'movies with runtime data:')
    print(movie_basics[movie_basics['start_year'] == year]['runtime_minutes'].isnull().value_counts())
    print('\n')

# of  2018 movies with runtime data:
False    12213
True      4636
Name: runtime_minutes, dtype: int64


# of  2019 movies with runtime data:
False    4501
True     3878
Name: runtime_minutes, dtype: int64


# of  2020 movies with runtime data:
True     855
False     82
Name: runtime_minutes, dtype: int64


# of  2021 movies with runtime data:
True     79
False     4
Name: runtime_minutes, dtype: int64


# of  2022 movies with runtime data:
True     29
False     3
Name: runtime_minutes, dtype: int64




So, exploring the IMDB SQLlite database, we see that this database includes records for FUTURE movies that have yet to be released! Furthermore, looking closer at the runtime data available, we can say there's probably only reliable movie data for the years 2010 to around 2019/2020. Judging by the fact that half the data from 2019 and almost all the data from 2020 onwards has no runtime data, this makes us assume that this database must have been extracted from IMDB sometime around 2019. Therefore, we need to look into how we can update this database with reliable data for movies released after 2019.
***

### Preliminary Data Exploration Summary:

In short, we've found our target variables and potential use-cases for the Box Office Mojo, TheNumbers, and the IMDB datasets. For each of these datasets, however, we've determined that they are all outdated! We must first update the data to make them current before we proceed with our analysis. The studio heads will not accept analysis based on old data!
***

## Finding New / Supplemental Data

In my search for up-to-date and current movie data, I was able to find (3) outside sources to supplement our datasets as follow:
> 1. movie_gross_data.csv : Maintained Box Office Mojo dataset; From Kaggle https://www.kaggle.com/
> 2. movie budgets.csv : Maintained comprehensive movie dataset including Budgets; From Kaggle https://www.kaggle.com/
> 3. Below files were the latest individual IMDB table downlaods. Data courtesy of IMDb. https://datasets.imdbws.com/
 
>* name.basics.tsv.gz
>* title.akas.tsv.gz
>* title.basics.tsv.gz
>* title.crew.tsv.gz
>* title.episode.tsv.gz
>* title.principals.tsv.gz
>* title.ratings.tsv.gz


Let's briefly introduce the new data sets, starting with Box Office Mojo dataset from Kaggle.

In [138]:
newbomdf = pd.read_csv("zippedData/movie_gross_data.csv", index_col=0)
newbomdf

Rank                                Release Group       Worldwide  \
0      1                            Top Gun: Maverick  $1,354,758,251   
1      2                      Jurassic World Dominion    $960,654,170   
2      3  Doctor Strange in the Multiverse of Madness    $954,860,995   
3      4                                   The Batman    $770,836,163   
4      5                     Minions: The Rise of Gru    $761,636,228   
..   ...                                          ...             ...   
30    31            Race for Your Life, Charlie Brown      $3,223,888   
31    32                                 March or Die      $1,000,000   
32    33                               Joseph Andrews        $839,865   
33    34                                Citizens Band        $815,530   
34    35                               Fraternity Row        $290,674   

        Domestic      %       Foreign    %.1  year  
0   $664,758,251  49.1%  $690,000,000  50.9%  2022  
1   $372,208,170  38.7%  $588,446,000  61.3%  2022  
2   $411,329,527  43.1%  $543,531,468  56.9%  2022  
3   $369,345,583  47.9%  $401,490,580  52.1%  2022  
4   $337,056,160  44.3%  $424,580,068  55.7%  2022  
..           ...    ...           ...    ...   ...  
30    $3,223,888   100%             -      -  1977  
31    $1,000,000   100%             -      -  1977  
32      $839,865   100%             -      -  1977  
33      $815,530   100%             -      -  1977  
34      $290,674   100%             -      -  1977  

[8440 rows x 8 columns]

Immediately, we see that this dataset has current movie data from this year! The dataset contains movie titles and years, along with box office financial performance data. Great news! We can use this! 
***
Let's look at the second next new data set from Kaggle.

In [141]:
budgets = pd.read_csv("zippedData/movies budgets.csv")
budgets

id                     title  \
0       766507                      Prey   
1       616037    Thor: Love and Thunder   
2       507086   Jurassic World Dominion   
3       438148  Minions: The Rise of Gru   
4       361743         Top Gun: Maverick   
...        ...                       ...   
755112  550108                      Live   
755113  549007                 Dagdromen   
755114  552103      Ekivoka Historietter   
755115  550704            To Say Goodbye   
755116  968161             Gising Sining   

                                           genres original_language  \
0          Action-Thriller-Science Fiction-Horror                en   
1                        Action-Adventure-Fantasy                en   
2                Adventure-Action-Science Fiction                en   
3       Family-Animation-Adventure-Comedy-Fantasy                en   
4                                    Action-Drama                en   
...                                           ...               ...   
755112                            Horror-Thriller                en   
755113                                      Drama                nl   
755114                                      Drama                xx   
755115                              Drama-Romance                zh   
755116                                        NaN                tl   

                                                 overview  popularity  \
0       When danger threatens her camp the fierce and ...    8775.992   
1       After his retirement is interrupted by Gorr th...    7709.008   
2       Four years after Isla Nublar was destroyed din...    4730.943   
3       A fanboy of a supervillain supergroup known as...    4499.308   
4       After more than thirty years of service as one...    3952.536   
...                                                   ...         ...   
755112  Linda Johnson's life is unraveling on Facebook...       0.600   
755113  Three high school teenagers reflect on the abs...       0.600   
755114  "Suggestive Short Stories" - The three episode...       0.600   
755115  We often fall in love first and then look for ...       0.600   
755116  In a country where fascism foolishness of peop...       0.600   

                                     production_companies release_date  \
0       Lawrence Gordon Productions-Davis Entertainmen...   2022-08-02   
1                  Marvel Studios-Kevin Feige Productions   2022-07-06   
2                 Universal Pictures-Amblin Entertainment   2022-06-01   
3           Universal Pictures-Illumination Entertainment   2022-06-29   
4       Paramount-Don Simpson/Jerry Bruckheimer Films-...   2022-05-24   
...                                                   ...          ...   
755112                                                NaN   2018-09-18   
755113                                Dutch Mountain Film   2018-10-02   
755114                                                NaN   1917-12-12   
755115                                  Da Huang Pictures   2009-01-01   
755116                                          Studio RD          NaN   

             budget       revenue  runtime    status  \
0               0.0  0.000000e+00    100.0  Released   
1       250000000.0  7.057000e+08    119.0  Released   
2       165000000.0  9.600000e+08    147.0  Released   
3        85000000.0  7.610000e+08     87.0  Released   
4       170000000.0  1.354000e+09    131.0  Released   
...             ...           ...      ...       ...   
755112          0.0  0.000000e+00     90.0  Released   
755113          0.0  0.000000e+00     10.0  Released   
755114          0.0  0.000000e+00     15.0  Released   
755115          0.0  0.000000e+00     20.0  Released   
755116          0.0  0.000000e+00     19.0  Released   

                                         tagline  vote_average  vote_count  \
0           They hunt to live. It lives to hunt.         8.198      2035.0   
1                       The one is 

We see that this dataset includes current movies with titles and years, but most importantly, contains useful budget data and genre data that were not present in the previous Box Office Mojo data set! It looks like if we combine this the two datasets together, we will have all the data we need to perform our financial analysis objective! More on that later!
***

Let's take a look at the new IMDB downloads starting with the basics table.

**NOTE: To clarify, the original IMDB database followed the ERD illustration. 
These new downloaded files are the individual table downloads of that previous ERD.
The tables still follow the same format/structure, but since they are no longer part of an SQL database, we've lost the powerful relational functionality between the tables! Unfortunately, we will have to find another way to relate the tables with one another!**

In [143]:
newtitles = pd.read_csv("zippedData/title.basics.tsv.gz", delimiter='\t')
newtitles

/opt/anaconda3/envs/learn-env/lib/python3.8/site-packages/IPython/core/interactiveshell.py:3145: DtypeWarning: Columns (4,5) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,


tconst  titleType               primaryTitle  \
0        tt0000001      short                 Carmencita   
1        tt0000002      short     Le clown et ses chiens   
2        tt0000003      short             Pauvre Pierrot   
3        tt0000004      short                Un bon bock   
4        tt0000005      short           Blacksmith Scene   
...            ...        ...                        ...   
9146674  tt9916848  tvEpisode              Episode #3.17   
9146675  tt9916850  tvEpisode              Episode #3.19   
9146676  tt9916852  tvEpisode              Episode #3.20   
9146677  tt9916856      short                   The Wind   
9146678  tt9916880  tvEpisode  Horrid Henry Knows It All   

                     originalTitle isAdult startYear endYear runtimeMinutes  \
0                       Carmencita       0      1894      \N              1   
1           Le clown et ses chiens       0      1892      \N              5   
2                   Pauvre Pierrot       0      1892      \N              4   
3                      Un bon bock       0      1892      \N             12   
4                 Blacksmith Scene       0      1893      \N              1   
...                            ...     ...       ...     ...            ...   
9146674              Episode #3.17       0      2010      \N             \N   
9146675              Episode #3.19       0      2010      \N             \N   
9146676              Episode #3.20       0      2010      \N             \N   
9146677                   The Wind       0      2015      \N             27   
9146678  Horrid Henry Knows It All       0      2014      \N             10   

                             genres  
0                 Documentary,Short  
1                   Animation,Short  
2          Animation,Comedy,Romance  
3                   Animation,Short  
4                      Comedy,Short  
...                             ...  
9146674         Action,Drama,Family  
9146675         Action,Drama,Family  
9146676         Action,Drama,Family  
9146677                       Short  
9146678  Adventure,Animation,Comedy  

[9146679 rows x 9 columns]

We immediately notice that these new IMDB downloads are going to be cumbersome due to the data sets having data for television shows and other non-movie media! We need to truncate this data before our computer explodes. 

In [145]:
movietitles = newtitles[newtitles['titleType'] == 'movie']
movietitles

tconst titleType  \
8        tt0000009     movie   
498      tt0000502     movie   
570      tt0000574     movie   
587      tt0000591     movie   
610      tt0000615     movie   
...            ...       ...   
9146569  tt9916622     movie   
9146596  tt9916680     movie   
9146608  tt9916706     movie   
9146619  tt9916730     movie   
9146629  tt9916754     movie   

                                              primaryTitle  \
8                                               Miss Jerry   
498                                               Bohemios   
570                            The Story of the Kelly Gang   
587                                       The Prodigal Son   
610                                     Robbery Under Arms   
...                                                    ...   
9146569        Rodolpho Teóphilo - O Legado de um Pioneiro   
9146596  De la ilusión al desconcierto: cine colombiano...   
9146608                                    Dankyavar Danka   
9146619                                             6 Gunn   
9146629                     Chico Albuquerque - Revelações   

                                             originalTitle isAdult startYear  \
8                                               Miss Jerry       0      1894   
498                                               Bohemios       0      1905   
570                            The Story of the Kelly Gang       0      1906   
587                                      L'enfant prodigue       0      1907   
610                                     Robbery Under Arms       0      1907   
...                                                    ...     ...       ...   
9146569        Rodolpho Teóphilo - O Legado de um Pioneiro       0      2015   
9146596  De la ilusión al desconcierto: cine colombiano...       0      2007   
9146608                                    Dankyavar Danka       0      2013   
9146619                                             6 Gunn       0      2017   
9146629                     Chico Albuquerque - Revelações       0      2013   

        endYear runtimeMinutes                      genres  
8            \N             45                     Romance  
498          \N            100                          \N  
570          \N             70  Action,Adventure,Biography  
587          \N             90                       Drama  
610          \N             \N                       Drama  
...         ...            ...                         ...  
9146569      \N             57                 Documentary  
9146596      \N            100                 Documentary  
9146608      \N             \N                      Comedy  
9146619      \N            116                          \N  
9146629      \N             49                 Documentary  

[618411 rows x 9 columns]

In [146]:
movietitles.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 618411 entries, 8 to 9146629
Data columns (total 9 columns):
 #   Column          Non-Null Count   Dtype 
---  ------          --------------   ----- 
 0   tconst          618411 non-null  object
 1   titleType       618411 non-null  object
 2   primaryTitle    618411 non-null  object
 3   originalTitle   618411 non-null  object
 4   isAdult         618411 non-null  object
 5   startYear       618411 non-null  object
 6   endYear         618411 non-null  object
 7   runtimeMinutes  618411 non-null  object
 8   genres          618411 non-null  object
dtypes: object(9)
memory usage: 47.2+ MB


Let's check out the principals table from the new IMDB data.

In [148]:
newprincipals = pd.read_csv("zippedData/title.principals.tsv.gz", delimiter='\t' )
newprincipals

tconst  ordering      nconst         category  \
0         tt0000001         1   nm1588970             self   
1         tt0000001         2   nm0005690         director   
2         tt0000001         3   nm0374658  cinematographer   
3         tt0000002         1   nm0721526         director   
4         tt0000002         2   nm1335271         composer   
...             ...       ...         ...              ...   
51607792  tt9916880         4  nm10535738          actress   
51607793  tt9916880         5   nm0996406         director   
51607794  tt9916880         6   nm1482639           writer   
51607795  tt9916880         7   nm2586970           writer   
51607796  tt9916880         8   nm1594058         producer   

                              job        characters  
0                              \N          ["Self"]  
1                              \N                \N  
2         director of photography                \N  
3                              \N                \N  
4                              \N                \N  
...                           ...               ...  
51607792                       \N  ["Horrid Henry"]  
51607793       principal director                \N  
51607794                       \N                \N  
51607795                    books                \N  
51607796                 producer                \N  

[51607797 rows x 6 columns]

We can immediately notice that these tables will not be useful until we develop a way to decipher the 'tconst' and 'nconst' columns. From the literature, we know these columns act as the 'movie title constant key' and 'person name constant key' respectively. This data will be critical for our Cast and Crew analysis. For now, we know have reliable new data from IMDB that is current and up to date.
***

## Strategy and Methodology

Now that we have explored and supplemented the data available for our analysis, we must strategize how and what we're going to do with each  set of data to achieve our objectives!



Our primary objective is to analyze the financial performance of modern movies. To do that, we stated that we are going to calculate the Profit/Loss(PnL) and Return-On-Investment(RoI) metrics for each movie. Let's define what those are mathematically:

> $PnL = (Total Worldwide Box Office Gross) - (Production Budget)$

> $RoI = (PnL)/(Production Budget)$

Now, in m

Box Office #'s and Production Budgets are the most important variables, and since The Numbers was the only data set to orignally contain both these variables, it makes sense to start with The Numbers dataset!

1. Merge The Numbers and Box Office Mojo datasets
2. Merge the supplemental movie budgets dataset
3. Clean Data
4. Calculate PnL and RoI metrics
7. Apply to account for "current"

In [ ]:
pd.read_sql("""
SELECT *
FROM movie_basics
LEFT JOIN movie_ratings USING(movie_id)
;
""", conn)

In [ ]:
pd.read_sql("""
SELECT *
FROM movie_ratings
;
""", conn)

In [ ]:
pd.read_sql("""
SELECT *
FROM movie_basics
LEFT JOIN movie_ratings USING(movie_id)
WHERE movie_id IN 
    (SELECT movie_id
    FROM principals
    WHERE person_id = "nm0000129")
;
""", conn)

In [ ]:
pd.read_sql("""
SELECT *
FROM movie_basics
WHERE movie_id IN 
    (SELECT movie_id
    FROM principals
    WHERE person_id = "nm0000129")
;
""", conn)

In [ ]:
pd.read_sql("""
SELECT *
FROM movie_basics
WHERE movie_id IN 
    (SELECT movie_id
    FROM principals
    WHERE person_id = "nm0000129")
;
""", conn)

In [ ]:
pd.read_sql("""
SELECT *
FROM persons;
""", conn)

In [ ]:
principals = pd.read_sql("""
SELECT *
FROM principals;
""", conn)

In [ ]:
cruiseknownfor = pd.read_sql("""
SELECT *
FROM known_for
WHERE person_id = "nm0000129";
""", conn)

In [ ]:
cruiseknownfor["movie_id"]

In [ ]:
pd.read_sql("""
SELECT *
FROM known_for
WHERE movie_id = "tt0325710";
""", conn)

In [ ]:
pd.read_sql("""
SELECT *
FROM principals
WHERE person_id = "nm0000129";
""", conn)

In [ ]:
pd.read_sql("""
SELECT *
FROM writers
WHERE person_id = "nm0000129";
""", conn)

In [ ]:
pd.read_sql("""
SELECT *
FROM persons
WHERE primary_name LIKE "%cruise%";
""", conn)

In [ ]:
pd.read_sql("""
SELECT *
FROM persons
WHERE primary_name LIKE "%jackman%";
""", conn)

In [ ]:
#Tom Cruise person_id = "nm0000129"

In [ ]:
pd.read_csv("zippedData/tmdb.movies.csv.gz",index_col=0).head(25)

In [ ]:
pd.read_csv("zippedData/tn.movie_budgets.csv.gz").head(50)

In [ ]:
pd.read_csv("zippedData/rt.movie_info.tsv.gz", delimiter='\t' ).head(25)

In [ ]:
pd.read_csv("zippedData/rt.reviews.tsv.gz", delimiter='\t', encoding='latin1').head(25)

In [ ]:
pd.read_csv("zippedData/rt.reviews.tsv.gz", delimiter='\t', encoding='latin1')

In [ ]:
pd.read_csv("zippedData/movie_gross_data.csv", index_col=0)

## Data Preparation

Describe and justify the process for preparing the data for analysis.

***
Questions to consider:
* Were there variables you dropped or created?
* How did you address missing values or outliers?
* Why are these choices appropriate given the data and the business problem?
***

In [ ]:
# Here you run your code to clean the data
thenumbers = pd.read_csv("zippedData/tn.movie_budgets.csv.gz")

In [ ]:
thenumbers.info()

In [ ]:
thenumbers.duplicated().values.sum() #No duplicates in dataset

In [ ]:
thenumbers["production_budget"].apply(type).value_counts()

In [ ]:
thenumbers["domestic_gross"].apply(type).value_counts()

In [ ]:
thenumbers["worldwide_gross"].apply(type).value_counts()

In [ ]:
thenumbers["production_budget"] = thenumbers["production_budget"].replace({'\$': '', ',': ''}, regex=True).astype(float)
thenumbers["production_budget"].apply(type).value_counts()

In [ ]:
thenumbers["domestic_gross"] = thenumbers["domestic_gross"].replace({'\$': '', ',': ''}, regex=True).astype(float)
thenumbers["domestic_gross"].apply(type).value_counts()

In [ ]:
thenumbers["worldwide_gross"] = thenumbers["worldwide_gross"].replace({'\$': '', ',': ''}, regex=True).astype(float)
thenumbers["worldwide_gross"].apply(type).value_counts()

In [ ]:
thenumbers["foreign_gross"] = thenumbers["worldwide_gross"] - thenumbers["domestic_gross"]

In [ ]:
thenumbers["foreign_gross"].apply(type).value_counts()

In [ ]:
thenumbers["profit_loss"] = thenumbers["worldwide_gross"] - thenumbers["production_budget"]

In [ ]:
thenumbers["profit_loss"].describe()

In [ ]:
thenumbers["roi"] = (thenumbers["profit_loss"] / thenumbers["production_budget"] ) *100

In [ ]:
thenumbers.sort_values(by='profit_loss', ascending=False).head(50).describe()

In [ ]:
thenumbers.sort_values(by='roi', ascending=False).head(50).describe()

In [ ]:
top500moviesbypnl = thenumbers.sort_values(by='profit_loss', ascending=False).head(500)["movie"]

In [ ]:
top500moviesbypnl

In [ ]:
top500moviesbyroi = thenumbers.sort_values(by='roi', ascending=False).head(500)["movie"]

In [ ]:
top500moviesbyroi

In [ ]:
#topmoviespnlroi = pd.Series(np.intersect1d(top500moviesbypnl,top500moviesbyroi))

topmoviespnlroi = pd.Series(list(set(top500moviesbypnl).intersection(set(top500moviesbyroi))))

topmoviespnlroi

In [ ]:
dfpnlroi = thenumbers[thenumbers["movie"].isin(topmoviespnlroi)]

In [ ]:
dfpnlroi.sort_values(by='profit_loss', ascending=False)

In [ ]:
dfpnlroi.loc[dfpnlroi['movie'] == 'Twilight']

In [ ]:
samenamemovies = dfpnlroi[dfpnlroi.duplicated('movie', keep=False)]

In [ ]:
samenamemovies.sort_values(by='profit_loss', ascending=False)

In [ ]:
samenamemovies.sort_values(by='roi', ascending=False)

In [ ]:
pd.plotting.scatter_matrix(samenamemovies, figsize=(15,15));

In [ ]:
samenamemovies.describe()

In [ ]:
dfpnlroi.describe()

In [ ]:
thenumbers.sort_values(by='profit_loss', ascending=False).head(500).describe() 
#minimum of top 500 pnl is 2.037891e+08 (Paddington)

In [ ]:
thenumbers.sort_values(by='roi', ascending=False).head(500)
#minimum of top 500 roi is 759.705987 (The Woman in Black)

In [ ]:
samenamemovies2 = samenamemovies[samenamemovies["profit_loss"] > 203789100]
samenamemovies2.sort_values(by = "profit_loss", ascending = False)
#samenamemovies5 = samenamemovies2[samenamemovies2["roi"] > 759.705987]
#samenamemovies5

In [ ]:
samenamemovies3 = samenamemovies[samenamemovies["roi"] > 759.705987]
samenamemovies3.sort_values(by = "roi", ascending = False)
#samenamemovies4 = samenamemovies3[samenamemovies3["profit_loss"] > 203789100]
#samenamemovies4

In [ ]:
worthymovies = pd.concat([samenamemovies2, samenamemovies3]).drop_duplicates()
worthymovies

In [ ]:
unworthymovies = samenamemovies[~samenamemovies.isin(worthymovies)].dropna()
unworthymovies

In [ ]:
dfpnlroi

In [ ]:
dfpnlroi2 = dfpnlroi[~dfpnlroi.isin(unworthymovies)].dropna()

In [ ]:
dfpnlroi2.describe()

In [ ]:
thenumbers.describe()

In [ ]:
dfpnlroi2['release_date'] = pd.to_datetime(dfpnlroi2['release_date'])
dfpnlroi2.sort_values(by="release_date", ascending=False).head(60)

In [ ]:
dfpnlroi2["release_date"].apply(type).value_counts()

In [ ]:
dfpnlroi2["release_date"].min()

In [ ]:
dfpnlroi2["release_date"].max()

In [ ]:
dfpnlroi[dfpnlroi['movie'].str.contains('mission', regex=False)]

In [ ]:
pd.read_csv("zippedData/tn.movie_budgets.csv.gz")

In [ ]:
bomdf = pd.read_csv("zippedData/bom.movie_gross.csv.gz")

In [ ]:
newbomdf = pd.read_csv("zippedData/movie_gross_data.csv", index_col=0)

In [ ]:
bomdf.info()

In [ ]:
bomdf.sort_values(by="year", ascending=False).head(60)

In [ ]:
newbomdf[newbomdf['Release Group'].str.contains('Mission', regex=False)]

In [ ]:
newbomdf.info()

In [ ]:
newbomdf.sort_values(by="year", ascending=False).head(60)

In [ ]:
budgets = pd.read_csv("zippedData/movies budgets.csv")

In [ ]:
justbudgets = budgets.drop(droplist,axis=1) #just extract budgets

In [ ]:
justbudgets.drop(justbudgets[justbudgets['budget'] <= 1].index, inplace = True) #remove records with no budget info

In [ ]:
justbudgets.drop_duplicates(inplace=True)

In [ ]:
justbudgets["movie (year)"] = justbudgets['title'].astype(str) + " (" + justbudgets['release_date'].str[0:4] + ")"

In [ ]:
justbudgets

In [ ]:
justbudgetsdup = justbudgets[justbudgets.duplicated(keep=False)]

In [ ]:
justbudgetsdup.head(50)

In [ ]:
#pd.merge(df1,df2,on ='Name', how ='left')

In [ ]:
budgets.info()

In [ ]:
droplist = list(budgets.columns)

In [ ]:
droplist.remove('title')

In [ ]:
droplist.remove('budget')

In [ ]:
droplist.remove('release_date')

In [ ]:
newbomdf.rename({'Release Group': 'title'}, axis=1, inplace=True)

In [ ]:
newbomdf.duplicated().values.sum()

In [ ]:
newbomdf

In [ ]:
newbomdf["movie (year)"] = newbomdf['title'].astype(str) + " (" + newbomdf['year'].astype(str).str[0:4] + ")"

In [ ]:
newbomdf[newbomdf['movie (year)'].str.contains('mission', regex=False)]

In [ ]:
newbomdf

In [ ]:
mergenewbombudgets = pd.merge(newbomdf,justbudgets,on ='movie (year)', how ='left')

In [ ]:
mergenewbombudgets

In [ ]:
mergenewbombudgets.dropna(inplace=True)

In [ ]:
mergenewbombudgets

In [ ]:
thenumbers #first time i did this, didn't create movie+year so had to do extra cool programming to deal with movies that share same name
#this time, tho, I will deal with this ahead of time

In [ ]:
thenumbers["movie (year)"] = thenumbers['movie'].astype(str) + " (" + thenumbers['release_date'].astype(str).str[-4:] + ")"

In [ ]:
thenumbers.head(50)

In [ ]:
thenumbers[thenumbers['movie (year)'].str.contains('Pirates', regex=False)]

In [ ]:
thenumbers['movie'] = thenumbers['movie'].str.replace("’","'")

In [ ]:
mergenewbombudgets[mergenewbombudgets['movie (year)'].str.contains('Mission', regex=False)]

In [ ]:
testing = pd.merge(mergenewbombudgets,thenumbers,on ='movie (year)', how ='outer')

In [ ]:
testing[["movie (year)","budget","production_budget"]]

In [ ]:
testing[["movie (year)","budget","production_budget"]]

In [ ]:
testing['maxBudget'] = testing.apply(
    lambda row: max(row["budget"] , row["production_budget"]), axis=1)

In [ ]:
testing['maxBudget'] = testing['maxBudget'].fillna(testing['production_budget'])

In [ ]:
testing.info()

In [ ]:
testing[["movie (year)","budget","production_budget", 'maxBudget']].head(2000)

In [ ]:
testing.head(4000)

In [ ]:
testing[testing['movie (year)'].str.contains('Mission', regex=False)]

In [ ]:
test = testing.drop(["Rank","%","%.1"], axis=1)

In [ ]:
test[~test.isnull().any(axis=1)]

In [ ]:
test[['Worldwide','worldwide_gross','movie (year)']]

In [ ]:
test["Worldwide"] = test["Worldwide"].replace({'\$': '', ',': ''}, regex=True).astype(float)
test["Worldwide"].apply(type).value_counts()

In [ ]:
test["Domestic"] = test["Domestic"].replace({'\$': '', ',': '', '-': '0'}, regex=True).astype(float)
test["Domestic"].apply(type).value_counts()

In [ ]:
test["Foreign"] = test["Foreign"].replace({'\$': '', ',': '', '-': '0'}, regex=True).astype(float)
test["Foreign"].apply(type).value_counts()

In [ ]:
test['WW UTD'] = test.apply(
    lambda row: max(row["Worldwide"] , row["worldwide_gross"]), axis=1)
test['WW UTD'] = test['WW UTD'].fillna(testing['worldwide_gross'])

In [ ]:
test[['Worldwide','worldwide_gross','WW UTD','movie (year)']].head(4800)

In [ ]:
test['DOM UTD'] = test.apply(
    lambda row: max(row["Domestic"] , row["domestic_gross"]), axis=1)
test['DOM UTD'] = test['DOM UTD'].fillna(testing['domestic_gross'])

In [ ]:
test['FOR UTD'] = test.apply(
    lambda row: max(row["Foreign"] , row["foreign_gross"]), axis=1)
test['FOR UTD'] = test['FOR UTD'].fillna(testing['foreign_gross'])

In [ ]:
test["profit_loss"] = test["WW UTD"] - test["maxBudget"]

In [ ]:
test["roi"] = (test["profit_loss"] / test["maxBudget"] ) *100

In [ ]:
test["Year"] = test['movie (year)'].str[-5:-1].astype(int)
test.info()

In [ ]:
test.drop("year2", axis=1, inplace=True)

In [ ]:
test.sort_values(by='maxBudget').head(500)

In [ ]:
test.drop(test[test['maxBudget'] < 1000000].index, inplace = True) 
#dropping all records with budget data less than $1mil as its causing skews

In [ ]:
test2 = test[test.columns[~test.isnull().any()]]

In [ ]:
test3 = test2.sort_values(by="Year", ascending = False).head(5000) 
#take 5000 most recent movies in database, cuz we want currently which equates to 1999

In [ ]:
test[test['movie (year)'].str.contains('Mission', regex=False)]

In [ ]:
test3roi500 = test3.sort_values(by="roi", ascending = False).head(1000)['movie (year)']

In [ ]:
test3pnl500 = test3.sort_values(by="profit_loss", ascending = False).head(500)['movie (year)']

In [ ]:
test3[test3['movie (year)'] == "Edge of Tomorrow (2014)"]

In [ ]:
test3[test3['movie (year)'] == "How to Train Your Dragon 2 (2014)"]

In [ ]:
test3pnl500.tail(6)

In [ ]:
test3[test3['movie (year)'] == "Crouching Tiger, Hidden Dragon (2000)"]

In [ ]:
test3pnlroi500 = pd.Series(list(set(test3roi500).intersection(set(test3pnl500))))

In [ ]:
test3pnlroi500.head(60)

In [ ]:
test3pnlroi500[test3pnlroi500 == "Edge of Tomorrow (2014)"]

In [ ]:
test4 = test3[test3["movie (year)"].isin(test3pnlroi500)]
test4

In [ ]:
test4.describe()

In [ ]:
test4[test4['movie (year)'].str.contains('Mission', regex=False)]

In [ ]:
test4.info()

In [ ]:
pd.set_option('display.max_rows', None)
test4

In [ ]:
pd.reset_option('display.max_rows')

In [ ]:
test4=test4.drop(2513)

In [ ]:
test4.describe()

In [ ]:
test4.sort_values(by="profit_loss", ascending=False).head(50)

In [ ]:
test4.sort_values(by="roi", ascending=False).head(50)

In [ ]:
test4["Profit/Loss ($, Millions)"] = round(test4['profit_loss']/1000000)

In [ ]:
test4["RoI (%)"] = round(test4['roi'])

In [ ]:
test4["Budget ($, Millions)"] = round(test4['maxBudget']/1000000)

In [ ]:
test4["Total Box Office ($, Millions)"] = round(test4['WW UTD']/1000000)

In [ ]:
test4

In [ ]:
test5 = test4[["movie (year)","Budget ($, Millions)","Total Box Office ($, Millions)","Profit/Loss ($, Millions)","RoI (%)","Year"]]

In [ ]:
test5.head(100)

## Data Modeling
Describe and justify the process for analyzing or modeling the data.

***
Questions to consider:
* How did you analyze or model the data?
* How did you iterate on your initial approach to make it better?
* Why are these choices appropriate given the data and the business problem?
***

In [ ]:
# Here you run your code to model the data
pd.plotting.scatter_matrix(test5, figsize=(20,20));

In [ ]:
plt.figure(figsize=(20,20))
sns.pairplot(test5,palette="husl")

In [ ]:
test5.describe()

In [ ]:
test5.cov()

In [ ]:
test5.corr(method='pearson')

In [ ]:
principals = pd.read_sql("""
SELECT pr.movie_id, pr.person_id, primary_name, primary_title, start_year
FROM principals AS pr
JOIN persons AS pe ON pr.person_id=pe.person_id
JOIN movie_basics AS mb ON pr.movie_id=mb.movie_id
;
""", conn)


In [ ]:
principals.loc[principals[principals["primary_title"].str.contains('Top Gun')].index, 'start_year'] = 2022

In [ ]:
principals["movie (year)"] = principals['primary_title'].astype(str) + " (" + principals['start_year'].astype(str).str[-4:] + ")"

In [ ]:
principals["movie (year)"].value_counts()

In [ ]:
#principals['start_year'] = principals[principals["primary_title"].str.contains('Top Gun')]['start_year'] == 2022

In [ ]:
principals[principals["primary_title"].str.contains('Top Gun')].index

In [ ]:
principals[principals["primary_title"].str.contains('Top Gun')]

In [ ]:
principals[principals["movie (year)"].isin(test5["movie (year)"])]["movie (year)"].value_counts()

In [ ]:
principals[principals["movie (year)"].isin(test5["movie (year)"])]["primary_name"].value_counts().head(50)

In [ ]:
principals[principals["primary_name"].str.contains('Cruise')]['movie (year)'].isin(test5['movie (year)'])

In [ ]:
pd.read_sql("""
SELECT pr.movie_id, pr.person_id, primary_name, primary_title, start_year
FROM principals AS pr
JOIN persons AS pe ON pr.person_id=pe.person_id
JOIN movie_basics AS mb ON pr.movie_id=mb.movie_id;
""", conn)


In [ ]:
pd.read_sql("""
SELECT *
FROM persons
WHERE primary_profession LIKE "%writer%"
;
""", conn)


In [ ]:
interesting = budgets[['title', 'genres','release_date','credits']]

In [ ]:
dropper = list(budgets.columns)

In [ ]:
dropper.remove('title')

In [ ]:
dropper.remove('genres')

In [ ]:
dropper.remove('release_date')

In [ ]:
dropper.remove('credits')

In [ ]:
interesting = budgets.drop(dropper,axis=1)

In [ ]:
interesting['movie (year)'] = interesting['title'].astype(str) + " (" + interesting['release_date'].astype(str).str[0:4] + ")"
interesting2 = interesting.dropna().drop_duplicates()

In [ ]:
interesting3 = interesting2[interesting2['movie (year)'].isin(test5['movie (year)'])]

In [ ]:
interesting3.head(60)

In [ ]:
interesting3["genre_split"] = interesting3['genres'].str.split("-")

In [ ]:
interesting4 = interesting3.explode("genre_split")

In [ ]:
interesting4

In [ ]:
#split in string of column

In [ ]:
newprincipals = pd.read_csv("zippedData/title.principals.tsv.gz", delimiter='\t' )

In [ ]:
pd.read_csv("zippedData/title.principals.tsv.gz", delimiter='\t' )

In [ ]:
newtitles = pd.read_csv("zippedData/title.basics.tsv.gz", delimiter='\t' )

In [ ]:
movietitles = newtitles[newtitles['titleType'] == 'movie']

In [ ]:
movietitles.drop(["isAdult","endYear",'runtimeMinutes'],axis=1,inplace = True)

In [ ]:
movietitles = movietitles[movietitles.startYear != '\\N']

In [ ]:
movietitles = movietitles[movietitles['startYear'].astype(int) > 1998]

In [ ]:
movietitles['movie (year)'] = movietitles['primaryTitle'].astype(str) + " (" + movietitles['startYear'].astype(str) + ")"
movietitles = movietitles.dropna().drop_duplicates()
movietitles.head(50)

In [ ]:
movietitles.drop(["titleType","primaryTitle",'originalTitle'],axis=1, inplace=True)

In [ ]:
moviekey = movietitles.set_index('tconst')
moviekey

In [ ]:
newprincipals

In [ ]:
newprincipals['movie (year)'] = newprincipals['tconst'].map(moviekey['movie (year)'])
newprincipals

In [ ]:
newprincipals['movi

In [ ]:
newprincipals['movie (year)'].value_counts()

In [ ]:
topprincipals = newprincipals[newprincipals['movie (year)'].isin(test5['movie (year)'])]

In [ ]:
topprincipals['movie (year)'].value_counts()

In [ ]:
#this may be interesting... the principals table only shows 10 principals per movie, so the max value of value_counts should be 10 for any movie
#this shows there are duplicate named movies! More investigation required on those 4 movies!

In [ ]:
topprincipals[topprincipals['movie (year)'].isin(
    ['Cinderella (2015)','Coco (2017)', 'Alice in Wonderland (2010)','Beauty and the Beast (2017)'])].tail(60)
#we see here that tconst tt5089556 is a duplicate bollywood  cinderella movie, DELETE
#tt11861230 is an imposter of B&B
#tt2049386 is an imposter of Alice
#tt7002100 is an imposter of Coco

In [ ]:
imposters = ['tt5089556' , 'tt11861230' ,'tt2049386' , 'tt7002100']

In [ ]:
topprincipals[topprincipals['tconst'].isin(imposters) == False]['movie (year)'].value_counts()
#this worked cleaning and removing imposters!

In [ ]:
newnames = pd.read_csv("zippedData/name.basics.tsv.gz", delimiter='\t' )

In [ ]:
newnames.head()

In [ ]:
newnames.drop_duplicates()

In [ ]:
namekey = newnames.set_index('nconst')

In [ ]:
namekey

In [ ]:
newnames[newnames['nconst'] == 'nm0000158']

In [ ]:
topprincipals['primary_name'] = topprincipals['nconst'].map(namekey['primaryName'])
topprincipals

In [ ]:
topprincipals['primary_name'].value_counts().head(50)

In [ ]:
top2principals = topprincipals[topprincipals['ordering'] < 3]

In [ ]:
top1principals = topprincipals[topprincipals['ordering'] < 2]

In [ ]:
top1principals['primary_name'].value_counts().head(50)

In [ ]:
topprincipals[topprincipals['primary_name'] == 'Mike Myers']

In [ ]:
test5[test5['movie (year)'].str.contains("Austin")]

In [ ]:
top1principals.head(50)

In [ ]:
top2principals.head(50)

In [ ]:
newcrew = pd.read_csv("zippedData/title.crew.tsv.gz", delimiter='\t' )

In [ ]:
newcrew.info()

In [ ]:
newcrew["director_split"] = newcrew['directors'].str.split(",")
newcrew2 = newcrew.explode("director_split")

In [ ]:
directorkey = newcrew2[["tconst","director_split"]]

In [ ]:
directorkey

In [ ]:
directorkey['movie (year)'] = directorkey['tconst'].map(moviekey['movie (year)'])
#directorkey['director'] = directorkey['nconst'].map(namekey['primaryName'])

In [ ]:
directorkey.loc[directorkey['movie (year)'] == 'Ted Bundy Had a Son (2022)']

In [ ]:
namekey

In [ ]:
directorkey['director'] = directorkey['director_split'].map(namekey['primaryName'])

In [ ]:
directorkey[directorkey['tconst'] == 'tt11861230']

In [ ]:
topdirectors = directorkey[directorkey['movie (year)'].isin(test5['movie (year)'])]

In [ ]:
topdirectors['movie (year)'].value_counts()

In [ ]:
topdirectors['director'].isnull().sum()

In [ ]:
topdirectors

In [ ]:
topdirectors[topdirectors['director'].isnull()]

In [ ]:
topdirectors.loc[2005836]

In [ ]:
topdirectors.head(60)

In [ ]:
topprincipals[topprincipals['category'] == 'director']['movie (year)'].value_counts()

In [ ]:
topdirectors['director'].value_counts().head(60)

In [ ]:
topdirectors['movie (year)'].value_counts()

In [ ]:
topprincipals.loc[topprincipals[''].duplicated()].head(60)
#important, this is how you find who is a co-director vs main director
#nope not here

In [ ]:
topdirectors.loc[topdirectors['movie (year)'].duplicated(keep=False)].head(60)

In [ ]:
topprincipals[topprincipals['ordering']==5]
#wow this is where I discovered ordering number 5 is always main director

In [ ]:
topprincipals[topprincipals['ordering']==5]['category'].value_counts()

In [ ]:
maindirectors = topprincipals[topprincipals['ordering']==5]

In [ ]:
maindirectors[maindirectors['movie (year)'].duplicated(keep=False)]
#found errors in data

In [ ]:
topprincipals[topprincipals['movie (year)'] == 'Cinderella (2015)']
#found that tconst tt5089556 is indian version of Cinderella (2015). DROP IT

In [ ]:
cleantopprincipals = topprincipals[topprincipals['tconst'].isin(imposters) == False]

In [ ]:
cleantopprincipals[cleantopprincipals['ordering']==5]

In [ ]:
maindirectors = cleantopprincipals[cleantopprincipals['ordering']==5]

In [ ]:
maindirectors.duplicated().sum()

In [ ]:
maindirectors['movie (year)'].value_counts()

In [ ]:
maindirectors[maindirectors['category'] != 'director']
#found problem where movies have directors who also acted in the movie!

In [ ]:
cleantopprincipals[cleantopprincipals['ordering']==5]['category'].value_counts()

In [ ]:
cleantopprincipals[cleantopprincipals['movie (year)']=='Top Gun: Maverick (2022)']

In [ ]:
onlymaindirectors = maindirectors[maindirectors['category'] == 'director']

In [ ]:
onlymaindirectors['primary_name'].value_counts()

In [ ]:
onlymaindirectors['movie (year)'].value_counts()

In [ ]:
cleantopprincipals.head(60)

In [ ]:
cleantopprincipals[cleantopprincipals['ordering']==7]['category'].value_counts()

In [ ]:
cleantopprincipals[cleantopprincipals['ordering']==9]['category'].value_counts()

In [ ]:
mainwriters = cleantopprincipals[cleantopprincipals['ordering'].isin([6,7])]

In [ ]:
onlymainwriters = mainwriters[mainwriters['category'] == 'writer']

In [ ]:
onlymainwriters['primary_name'].value_counts()

In [ ]:
onlymainwriters['movie (year)'].value_counts()

In [ ]:
onlymainwriters.head(60)

In [ ]:
onlymainwriters.drop_duplicates(subset='movie (year)', keep='first', inplace=True )

In [ ]:
onlymainwriters['movie (year)'].value_counts()

In [ ]:
onlymainwriters['primary_name'].value_counts()

In [ ]:
top3principals = cleantopprincipals[cleantopprincipals['ordering'] < 4]

In [ ]:
top3males = top3principals[top3principals['category'] == 'actor']

In [ ]:
top3females = top3principals[top3principals['category'] == 'actress']

In [ ]:
top3males['movie (year)'].value_counts()

In [ ]:
top3females['movie (year)'].value_counts()

In [ ]:
top3principals['category'].value_counts()

In [ ]:
top2males['primary_name'].value_counts().head(60)

In [ ]:
toplead = cleantopprincipals[cleantopprincipals['ordering'] < 2]

In [ ]:
topmale = toplead[toplead['category'] == 'actor']

In [ ]:
topfemale = toplead[toplead['category'] == 'actress']

In [ ]:
topmale['primary_name'].value_counts().head(60)

In [ ]:
topfemale['primary_name'].value_counts()

In [ ]:
topfemale['movie (year)'].value_counts()

In [ ]:
top4females

In [ ]:
top4females.drop_duplicates(subset='movie (year)', keep='first', inplace=True )

In [ ]:
top4females['movie (year)'].value_counts()

In [ ]:
top4males['movie (year)'].value_counts()

In [ ]:
top4males.drop_duplicates(subset='movie (year)', keep='first', inplace=True )

In [ ]:
top4males['primary_name'].value_counts()

In [ ]:
top4females['primary_name'].value_counts()

In [ ]:
%whos

In [ ]:
test5

In [ ]:
omdmerge = onlymaindirectors[["movie (year)","primary_name"]]

In [ ]:
omdmerge.rename(columns ={'primary_name':'director'}, inplace = True)


In [ ]:
omdmerge.set_index('movie (year)')

In [ ]:
test6 = pd.merge(test5, omdmerge, on ='movie (year)',how ='left')

In [ ]:
test6.groupby('director').mean().sort_values(by = 'Total Box Office ($, Millions)',ascending=False).head(5)

In [ ]:
omwmerge = onlymainwriters[["movie (year)","primary_name"]]
omwmerge.rename(columns ={'primary_name':'writer'}, inplace = True)
omwmerge.set_index('movie (year)')

In [ ]:
test7 = pd.merge(test6, omwmerge, on ='movie (year)',how ='left')

In [ ]:
test7.groupby('writer').mean().sort_values(by = 'Total Box Office ($, Millions)',ascending=False).head(5)

In [ ]:
malemerge = top4males[["movie (year)","primary_name"]]
malemerge.rename(columns ={'primary_name':'lead actor'}, inplace = True)
malemerge.set_index('movie (year)')

In [ ]:
femalemerge = top4females[["movie (year)","primary_name"]]
femalemerge.rename(columns ={'primary_name':'lead actress'}, inplace = True)
femalemerge.set_index('movie (year)')

In [ ]:
test8 = pd.merge(test7, malemerge, on ='movie (year)',how ='left')

In [ ]:
test9 = pd.merge(test8, femalemerge, on ='movie (year)',how ='left')

In [ ]:
test9.loc[test9['lead actor'] == 'Leonardo DiCaprio']

In [ ]:
test9.groupby('lead actor').mean().sort_values(by = 'Profit/Loss ($, Millions)',ascending=False).head(50)

In [ ]:
test9.groupby('lead actor').mean().sort_values(by = 'RoI (%)',ascending=False).head(50)

In [ ]:
test9['lead actor'].value_counts()

In [ ]:
test9['lead actress'].value_counts().head(25)

In [ ]:
test9[test9['lead actor'] == 'Tom Cruise']

In [ ]:
test9.describe()

In [ ]:
test9[test9["movie (year)"] == 'Crouching Tiger, Hidden Dragon (2000)']

In [ ]:
interesting4[['movie (year)','genre_split']]

In [ ]:
interesting3

In [ ]:
genremerge = interesting3[["movie (year)","genre_split"]]
genremerge.set_index('movie (year)')
genremerge

In [ ]:
testgenre = pd.merge(test5, genremerge, on ='movie (year)',how ='left')
testgenre

In [ ]:
testgenre2 = testgenre.explode("genre_split")
testgenre2

In [ ]:
testgenre2['genre_split'].value_counts()

In [ ]:
genrecountdata

In [ ]:
plt.figure(figsize=(18,20))
# Draw the seaborn barplot
sns.countplot(y='genre_split', data=testgenre2 )
# Set the barplot's title.
plt.title('Genre Counts of Top Movie Pool')
# Label the y-axis
plt.ylabel('Genre')
# Label the x-axis
plt.xlabel('Count')

In [ ]:
testgenre2.groupby('genre_split').mean().sort_values(by = 'Profit/Loss ($, Millions)',ascending=False)

In [ ]:
testgenre2.groupby('genre_split').mean().sort_values(by = 'RoI (%)',ascending=False)

In [ ]:
testgenre3 = testgenre2.groupby('genre_split').mean().sort_values(by = 'Profit/Loss ($, Millions)',ascending=False)

In [ ]:
testgenre4 = testgenre3.reset_index()

In [ ]:
testgenre5 = testgenre2.groupby('genre_split').mean().sort_values(by = 'RoI (%)',ascending=False).reset_index()

In [ ]:
testgenre4

In [ ]:
test5

In [ ]:
# not needed dont use

sns.set(style="whitegrid")
# Set the scale plot to be larger and easier to read.
sns.set_context("talk")
# Create a plot and size it appropriately for information shown
plt.figure(figsize=(15,12))
# Draw the seaborn barplot
sns.barplot(y = keys, x = values, alpha=0.8, palette="deep")
# Set the barplot's title.
plt.title('Top 100 grossing films by Genre')
# Label the y-axis
plt.ylabel('Genre', fontsize=20)
# Label the x-axis
plt.xlabel('Count', fontsize=20)

In [ ]:
# Create a plot and set the appropriate size
plt.figure(figsize=(15, 12))
# Draw a seaborn scatterplot based on worldwide gross and production budget,

sns.scatterplot(x=test5['Budget ($, Millions)'],
                y=test5['Total Box Office ($, Millions)'], alpha=0.6, palette='blues')

#sns.regplot(x=test5['Budget ($, Millions)'], y=test5['Total Box Office ($, Millions)'], line_kws={"color":"r","alpha":0.7})

# Set title of plot
plt.title('Box Office VS Production Budgets')
# Set y-axis label and fontsize
plt.ylabel('Total Box Office ($, Millions)')

# Set x-axis label and fontsize
plt.xlabel('Budget ($, Millions)')

# Change x-axis ticks from scientific notation to integers
#plt.ticklabel_format(style='plain', axis='x')
# Change y-axis ticks from scientific notation to integers
#plt.ticklabel_format(style='plain', axis='y')
# Show the plot
plt.show()

In [ ]:
# Create a plot and set the appropriate size
plt.figure(figsize=(15, 12))
# Draw a seaborn scatterplot based on worldwide gross and production budget,

sns.scatterplot(x=test5['Budget ($, Millions)'],
                y=test5['RoI (%)'], alpha=0.6, palette='blues')
# Set title of plot
plt.title('RoI VS Production Budgets')
# Set y-axis label and fontsize
plt.ylabel('RoI (%)')

# Set x-axis label and fontsize
plt.xlabel('Budget ($, Millions)')

# Change x-axis ticks from scientific notation to integers
#plt.ticklabel_format(style='plain', axis='x')
# Change y-axis ticks from scientific notation to integers
#plt.ticklabel_format(style='plain', axis='y')
# Show the plot
plt.show()

In [ ]:
# Create a plot and set the appropriate size
plt.figure(figsize=(15, 12))
# Draw a seaborn scatterplot based on worldwide gross and production budget,

sns.scatterplot(x=test5['Budget ($, Millions)'],
                y=test5['Profit/Loss ($, Millions)'], alpha=0.6, palette='blues')
# Set title of plot
plt.title('Profit/Loss VS Production Budgets')
# Set y-axis label and fontsize
plt.ylabel('Profit/Loss ($, Millions)')

# Set x-axis label and fontsize
plt.xlabel('Budget ($, Millions)')

# Change x-axis ticks from scientific notation to integers
#plt.ticklabel_format(style='plain', axis='x')
# Change y-axis ticks from scientific notation to integers
#plt.ticklabel_format(style='plain', axis='y')
# Show the plot
plt.show()

In [ ]:
# Create a plot and set the appropriate size
plt.figure(figsize=(15, 12))
# Draw a seaborn scatterplot based on worldwide gross and production budget,

sns.histplot(x=test5['Budget ($, Millions)'],
                y=test5['Profit/Loss ($, Millions)'], alpha=0.6, palette='blues')
# Set title of plot
plt.title('Profit/Loss VS Production Budgets')
# Set y-axis label and fontsize
plt.ylabel('Profit/Loss ($, Millions)')

# Set x-axis label and fontsize
plt.xlabel('Budget ($, Millions)')

# Change x-axis ticks from scientific notation to integers
#plt.ticklabel_format(style='plain', axis='x')
# Change y-axis ticks from scientific notation to integers
#plt.ticklabel_format(style='plain', axis='y')
# Show the plot
plt.show()

In [ ]:
# Create a plot and set the appropriate size
plt.figure(figsize=(15, 12))
# Draw a seaborn scatterplot based on worldwide gross and production budget,

sns.scatterplot(x='Budget ($, Millions)',y='Profit/Loss ($, Millions)', data=testgenre4, hue='genre_split', s=150)
# Set title of plot
plt.title('Profit/Loss VS Production Budgets')
# Set y-axis label and fontsize
plt.ylabel('Profit/Loss ($, Millions)')

# Set x-axis label and fontsize
plt.xlabel('Budget ($, Millions)')

# Change x-axis ticks from scientific notation to integers
#plt.ticklabel_format(style='plain', axis='x')
# Change y-axis ticks from scientific notation to integers
#plt.ticklabel_format(style='plain', axis='y')
# Show the plot
plt.show()

In [ ]:
# Create a plot and set the appropriate size
plt.figure(figsize=(15, 12))
# Draw a seaborn scatterplot based on worldwide gross and production budget,

sns.scatterplot(x='Budget ($, Millions)',y='RoI (%)', data=testgenre5, hue='genre_split', s=150)
# Set title of plot
plt.title('RoI (%) VS Production Budgets')
# Set y-axis label and fontsize
plt.ylabel('RoI (%)')

# Set x-axis label and fontsize
plt.xlabel('Budget ($, Millions)')

# Change x-axis ticks from scientific notation to integers
#plt.ticklabel_format(style='plain', axis='x')
# Change y-axis ticks from scientific notation to integers
#plt.ticklabel_format(style='plain', axis='y')
# Show the plot
plt.show()

In [ ]:
malemetrics = test9.groupby('lead actor').mean()

In [ ]:
malemetrics.loc['Tom Cruise']

In [ ]:
top10maleVC = test9['lead actor'].value_counts().head(14)

In [ ]:
top10maleVClist = top10maleVC.index.tolist()

In [ ]:
malemetrics.loc[top10maleVClist]

In [ ]:
maledata = malemetrics.loc[top10maleVClist].reset_index()

In [ ]:
# Create a plot and set the appropriate size
plt.figure(figsize=(15, 12))
# Draw a seaborn scatterplot based on worldwide gross and production budget,

sns.scatterplot(y='Profit/Loss ($, Millions)',x='RoI (%)', data=maledata, hue='lead actor', s=150)
# Set title of plot
plt.title('RoI (%) Profit/Loss ($, Millions)')
# Set y-axis label and fontsize
plt.ylabel('Profit/Loss ($, Millions)')

# Set x-axis label and fontsize
plt.xlabel('RoI (%)')

# Change x-axis ticks from scientific notation to integers
#plt.ticklabel_format(style='plain', axis='x')
# Change y-axis ticks from scientific notation to integers
#plt.ticklabel_format(style='plain', axis='y')
# Show the plot
plt.show()

In [ ]:
# Create a plot and define its size.
plt.figure(figsize=(20, 12))
# Draw a boxplot showing the runtime minute distributions among the ranges of
# production budget.
sns.boxplot(x='lead actor',
            y='Profit/Loss ($, Millions)',data=maleboxdata)
# Set the plot title
plt.title('Distribution of Profit by Lead Actor')
# Set the x-axis Label and define fontsize
plt.xlabel('Lead Actor')
# Set the y-axis label and define fontsize
plt.ylabel('Profit/Loss ($, Millions)')

# Show plot
plt.show()

In [ ]:
maleboxdata = test9[test9['lead actor'].isin(top10maleVClist)]

In [ ]:
maleboxdata

In [ ]:
#take the top 10 recurring billed actresses from top movie pool
top10femaleVC = test9['lead actress'].value_counts().head(10)

#turn series into list
top10femaleVClist = top10femaleVC.index.tolist()

#create averages data for scatter plot
femalemetrics = test9.groupby('lead actress').mean()
femaledata = femalemetrics.loc[top10femaleVClist].reset_index()

#create dataframe slice for boxplot
femaleboxdata = test9[test9['lead actress'].isin(top10femaleVClist)]



In [ ]:
# Create a plot and set the appropriate size
plt.figure(figsize=(15, 12))
# Draw a seaborn scatterplot based on worldwide gross and production budget,

sns.scatterplot(y='Profit/Loss ($, Millions)',x='RoI (%)', data=femaledata, hue='lead actress', s=150)
# Set title of plot
plt.title('RoI (%) Profit/Loss ($, Millions)')
# Set y-axis label and fontsize
plt.ylabel('Profit/Loss ($, Millions)')

# Set x-axis label and fontsize
plt.xlabel('RoI (%)')

# Change x-axis ticks from scientific notation to integers
#plt.ticklabel_format(style='plain', axis='x')
# Change y-axis ticks from scientific notation to integers
#plt.ticklabel_format(style='plain', axis='y')
# Show the plot
plt.show()

In [ ]:
# Create a plot and define its size.
plt.figure(figsize=(15, 12))
# Draw a boxplot showing the runtime minute distributions among the ranges of
# production budget.
sns.boxplot(x='lead actress',
            y='Profit/Loss ($, Millions)',data=femaleboxdata)
# Set the plot title
plt.title('Distribution of Profit by Lead Actress')
# Set the x-axis Label and define fontsize
plt.xlabel('Lead Actress')
# Set the y-axis label and define fontsize
plt.ylabel('Profit/Loss ($, Millions)')

# Show plot
plt.show()

In [ ]:
test9['director'].value_counts().head(40)

In [ ]:
#take the top averaging directors from top movie pool
#top10dirVC = test9['director'].value_counts().head(10)

#turn series into list
#top10dirVClist = top10dirVC.index.tolist()

#create averages data for scatter plot, take top 10
dirmetrics = test9.groupby('director').mean().sort_values(by = 'Profit/Loss ($, Millions)',ascending=False).head(10)
dirdata = dirmetrics.reset_index()

#take above top 10 series into list
top10dirVClist = dirmetrics.index.tolist()

#create dataframe slice for boxplot
dirboxdata = test9[test9['director'].isin(top10dirVClist)]


In [ ]:
# Create a plot and set the appropriate size
plt.figure(figsize=(15, 12))
# Draw a seaborn scatterplot based on worldwide gross and production budget,

sns.scatterplot(y='Profit/Loss ($, Millions)',x='RoI (%)', data=dirdata, hue='director', s=150)
# Set title of plot
plt.title('RoI (%) Profit/Loss ($, Millions)')
# Set y-axis label and fontsize
plt.ylabel('Profit/Loss ($, Millions)')

# Set x-axis label and fontsize
plt.xlabel('RoI (%)')

# Change x-axis ticks from scientific notation to integers
#plt.ticklabel_format(style='plain', axis='x')
# Change y-axis ticks from scientific notation to integers
#plt.ticklabel_format(style='plain', axis='y')
# Show the plot
plt.show()

In [ ]:
# Create a plot and define its size.
plt.figure(figsize=(15, 12))
# Draw a boxplot showing the runtime minute distributions among the ranges of
# production budget.
sns.boxplot(x='director',
            y='Profit/Loss ($, Millions)',data=dirboxdata)
# Set the plot title
plt.title('Distribution of Profit by Director')
# Set the x-axis Label and define fontsize
plt.xlabel('Director')
# Set the y-axis label and define fontsize
plt.ylabel('Profit/Loss ($, Millions)')

# Show plot
plt.show()

In [ ]:
#create averages data for scatter plot, take top 10
wrimetrics = test9.groupby('writer').sum().sort_values(by = 'Total Box Office ($, Millions)',ascending=False).head(10)
wridata = wrimetrics.reset_index()

#take above top 10 series into list
top10wriVClist = wrimetrics.index.tolist()

#create dataframe slice for boxplot
wriboxdata = test9[test9['writer'].isin(top10wriVClist)]

In [ ]:
test9['writer'].value_counts().head(40)

In [ ]:
# Create a plot and set the appropriate size
plt.figure(figsize=(18, 12))
# Draw a seaborn scatterplot based on worldwide gross and production budget,

sns.barplot(y='Total Box Office ($, Millions)',x='writer', data=wridata)
# Set title of plot
plt.title('Writer Cumulative Box Office')
# Set y-axis label and fontsize
plt.ylabel('Total Box Office ($, Millions)')

# Set x-axis label and fontsize
plt.xlabel('Writer')

# Change x-axis ticks from scientific notation to integers
#plt.ticklabel_format(style='plain', axis='x')
# Change y-axis ticks from scientific notation to integers
#plt.ticklabel_format(style='plain', axis='y')
# Show the plot
plt.show()

In [ ]:
# Create a plot and define its size.
plt.figure(figsize=(20, 12))
# Draw a boxplot showing the runtime minute distributions among the ranges of
# production budget.
sns.boxplot(x='writer',
            y='Profit/Loss ($, Millions)',data=wriboxdata)
# Set the plot title
plt.title('Distribution of Profit by Writer')
# Set the x-axis Label and define fontsize
plt.xlabel('Writer')
# Set the y-axis label and define fontsize
plt.ylabel('Profit/Loss ($, Millions)')

# Show plot
plt.show()

## Conclusions
Provide your conclusions about the work you've done, including any limitations or next steps.

***
Questions to consider:
* What would you recommend the business do as a result of this work?
* What are some reasons why your analysis might not fully solve the business problem?
* What else could you do in the future to improve this project?
***

In [ ]:
#Target Production Budget Sweet Spot

In [ ]:
#Recommended Cast and Crew

In [ ]:
#Recommended Genre Type